web/components/tipper.tsx

```diff
@@ -0,0 +1,160 @@
+/* eslint-disable react-hooks/exhaustive-deps */
```

Comment:
It's better not to use this if you can. It looks pretty straightforward to add the correct dependencies to your effects here.